messaging/src/main/java/org/axonframework/eventhandling/ReplayStatus.java

```diff
@@ -21,8 +21,9 @@
  * a replay, or in regular operations. Messages delivered as part of a replay may have been handled by this handler
  * before.
  * <p>
- * Note that this is only sensible for event handlers that are assigned to a Tracking Processor. Event Handlers assigned
- * to another type of processor will only receive events in "Regular" operation.
+ * Note that this is only sensible for event handlers that are assigned to a {@link StreamingEventProcessor}. Event
+ * Handlers assigned to a {@link SubscribingEventProcessor}, which is immediately invoked during command handling, will
```

Comment:
Structly speaking, subscribing processors aren't "invoked directly during command handling". Subscribing processors rely on the threads of the publishing mechanism to deliver the events. Thefore, whether a message is a replay or not isn't controlled by the processor, but rather by whichever mechanism publishes it. None of the current implementations support replays, but it's not impossible to have a subscribing processor that supports replays...

Reply:
Ah right, I'll remove the command handling part. I don't get the "current implementations" part of SubscribingEventProcessors though, since it's a concrete implementation of `AbstractEventProcessor` directly. What would you propose here?

Reply:
With "current implementations", I mean the current implementations of mechanisms that publish the events to a subscribing processor. Besides the local event bus segment, this could also be the Kafka extension. While Kafka could possibly be triggered to replay all events, the extension currently doesn't indicate whether an event it publishes is a "replay" or an event that was passed for the first time...

Reply:
Okay, I tried to capture that as follows:

```
 * Note that this is only sensible for event handlers that are assigned to a {@link StreamingEventProcessor}. Event
 * Handlers assigned to a different mechanism, such as the {@link SubscribingEventProcessor} or certain publishing methods of extensions,
 * will always receive {@code ReplayStatus.REGULAR} as value.
 ```